lib/checkclass.cpp

```diff
@@ -686,6 +686,21 @@ void CheckClass::assignAllVar(std::vector<Usage> &usageList)
         i.assign = true;
 }
 
+void CheckClass::assignAllVarsVisibleFromScope(std::vector<Usage>& usageList, const Scope* scope)
+{
+    for (Usage& usage : usageList) {
+        if (usage.var->scope() == scope)
+            usage.assign = true;
+    }
+
+    // Iterate through each base class...
+    for (const Type::BaseInfo& i : scope->definedType->derivedFrom) {
+        const Type *derivedFrom = i.type;
+
+        assignAllVarsVisibleFromScope(usageList, derivedFrom->classScope);
```

Comment:
Looks like `derivedFrom` needs to be checked for `nullptr`, see http://cppcheck1.osuosl.org:8000/crash.html